src/library_fs.js

```diff
@@ -213,11 +219,17 @@ FS.staticInit();
         // setting opts.follow = true will override this behavior.
         if (!islast || opts.follow) {
           var count = 0;
-          while (FS.isLink(current.mode)) {
-            var link = FS.readlink(current_path);
+          while (current && FS.isLink(current.mode)) {
+            if (!current.node_ops.readlink) {
+              throw new FS.ErrnoError({{{ cDefs.ENOSYS }}});
+            }
+            var link = current.node_ops.readlink(current);
```

Comment:
Before this change, `FS.readlink` calls `lookupPath` again which repeats exactly the same work we've done up to this point in this function except with `opts.follow` false. Rather than doing that, we already have the node here so I figure it's better just to call `node_ops.readlink`.